presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -160,137 +162,139 @@ public void setCreateTableAsSelectNoOp(boolean createTableAsSelectNoOp)
 
     public void setAggregates(QuerySpecification node, List<FunctionCall> aggregates)
     {
-        this.aggregates.put(node, aggregates);
+        this.aggregates.put(NodeRef.of(node), aggregates);
     }
 
     public List<FunctionCall> getAggregates(QuerySpecification query)
     {
-        return aggregates.get(query);
+        return aggregates.get(NodeRef.of(query));
     }
 
     public void setOrderByAggregates(OrderBy node, List<Expression> aggregates)
     {
-        this.orderByAggregates.put(node, ImmutableList.copyOf(aggregates));
+        this.orderByAggregates.put(NodeRef.of(node), ImmutableList.copyOf(aggregates));
     }
 
-    public List<Expression> getOrderByAggregates(OrderBy query)
+    public List<Expression> getOrderByAggregates(OrderBy node)
     {
-        return orderByAggregates.get(query);
+        return orderByAggregates.get(NodeRef.of(node));
     }
 
     public IdentityLinkedHashMap<Expression, Type> getTypes()
     {
-        return new IdentityLinkedHashMap<>(types);
+        return NodeRefCollections.toIdentityMap(types);
```

Comment:
I wonder whether you though about using some view on `Map<NodeRef<? extends Node>, T>` instead? This way we would avoid copying on each getter call. Otherwise probably we should think of making this method depricated maybe.

Quick search: You could use `https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/ForwardingMap.html` to provide identity based view on `Map<NodeRef<? extends Node>, T>`.